Visual Studio 2019 debugger reports error

**Describe the bug**
When instantiating a new instance of the Patient class, and debugging, an error message is shown.  The error message is: {"Patient/null" Identity=error CS0103: The name 'ResourceIdentity' does not exist in the current context}

The error occurs when using version 2.0.1, but not when using version 1.9.0.

The error has also been seen for other entities such as Encounter and Observation

**To Reproduce**
Steps to reproduce the behavior:
1. Create a new .NET core 3.1 console app. 
2. Install version 2.0.1 of hl7.fhir.r4 from Nuget
2. Write code to instantiate a new Patient
3. Debug
4. In the Locals window see the error CS0103

**Expected behavior**
No error shown while debugging

**Screenshots**
Screenshot added that shows the difference between version 1.9.0 and 2.0.1

**Version used:**
 - FHIR Version: R4
 - Version:  2.0.1
![image](https://user-images.githubusercontent.com/3274517/105504621-45407980-5cc8-11eb-9b7e-c51c8fa3e6b7.png)

@marcovisserFurore this bug seems to be a little more serious than expected. It has broken our FHIR server in regards to linking / bundling resources to a patient resource. 

,,,
var client = new FhirClient(configuration["resource"]);
                var patientId = reqdata.patientid;
                FhirToken fhToken = new FhirToken(configuration);
                var bearerToken = fhToken.GetBearerToken();
                client.OnBeforeRequest += (object sender, BeforeRequestEventArgs e) =>
                {
                    e.RawRequest.Headers.Add("Authorization", string.Format("Bearer {0}", bearerToken));
                };
 ,,,
 
New updated client creation code as below which you can find in feature/FHIR2 branch.
 
,,,
FhirToken fhToken = new FhirToken(configuration);
                var bearerToken = fhToken.GetBearerToken();
                var messageHandler = new HttpClientEventHandler();
                messageHandler.OnBeforeRequest += (object sender, BeforeHttpRequestEventArgs e) =>
                {
                    e.RawRequest.Headers
                    .Add("Authorization", $"Bearer {bearerToken}");
                };
 
                Hl7.Fhir.Rest.FhirClient client = new Hl7.Fhir.Rest.FhirClient(configuration["resource"], messageHandler: messageHandler, settings: new FhirClientSettings()
                {
                    PreferredFormat = ResourceFormat.Json
                });

,,,
 
Now there is a failure in creating a reference between patient resource and any other resource like (diagnosisreport/Obervance/Vitals etc..) with an error
Identity=error CS0103: The name 'ResourceIdentity' does not exist in the current context
 
`var diagnosereport = new DiagnosticReport();`
                     
`diagnosereport.Subject = new ResourceReference(string.Format("Patient/{0}", fhirPatientId));`

I installed latest version of Fhirly and even tried with downgraded version but i am getting same error. [0] = {"Patient/null" Identity=error CS0103: The name 'ResourceIdentity' does not exist in the current context}. I Removed resource identity still same error.
![image](https://user-images.githubusercontent.com/78563502/106946363-00b8d300-66f7-11eb-99da-5318f9d45bc0.png)
this is my json

[
   {
      "active": true,
      "telecom": [
         {
            "value": "9910909654",
            "system": "phone"
         }
      ],
      "resourceType": "Patient",
      "birthDate": "1990-02-13",
      "meta": {
         "versionId": "2"
      },
      "gender": "female",
      "name": [
         {
            "prefix": [
               "NULL"
            ],
            "given": [
               "Joelle"
            ],
            "text": "Joelle Sammy",
            "family": "Sammy"
         }
      ],
      "address": [
         {
            "postalCode": "10008",
            "text": "NULL",
            "state": "XX",
            "line": [
               "Street10008"
            ],
            "city": "City10008",
            "country": "NULL"
         }
      ],
      "id": "456"
   },
   {
      "active": true,
      "telecom": [
         {
            "value": "9910909630",
            "system": "phone"
         }
      ],
      "resourceType": "Patient",
      "birthDate": "1987-06-14",
      "meta": {
         "versionId": "3"
      },
      "gender": "male",
      "name": [
         {
            "prefix": [
               "MR"
            ],
            "given": [
               "Joey"
            ],
            "text": "Joey Sammuel",
            "family": "Sammuel"
         }
      ],
      "address": [
         {
            "postalCode": "10007",
            "text": "NULL",
            "state": "XX",
            "line": [
               "Street10007"
            ],
            "city": "City10007",
            "country": "NULL"
         }
      ],
      "id": "789"
   }
]

--- a/src/Hl7.Fhir.Core/Model/StructureDefinition.cs
+++ b/src/Hl7.Fhir.Core/Model/StructureDefinition.cs
@@ -26,13 +26,12 @@
   POSSIBILITY OF SUCH DAMAGE.
   
 */
-
-using System;
 using System.Collections.Generic;
 using System.ComponentModel;
 using Hl7.Fhir.Introspection;
 using Hl7.Fhir.Validation;
 using System.Linq;
+using System.Diagnostics;
 
 namespace Hl7.Fhir.Model
 {
@@ -43,9 +42,12 @@ namespace Hl7.Fhir.Model
     }
 
     // [WMR 20161005] Added specific debugger display attribute that includes the canonical url
-    [System.Diagnostics.DebuggerDisplay("\\{\"{TypeName,nq}/{Id,nq}\" Identity={ResourceIdentity()}} Url={Url}")]
+    [System.Diagnostics.DebuggerDisplay("\\{\"{TypeName,nq}/{Id,nq}\" Identity={DebuggerDisplay}} Url={Url}")]
     public partial class StructureDefinition
     {
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+        private Rest.ResourceIdentity DebuggerDisplay => this.ResourceIdentity();
+       
         public partial class SnapshotComponent : IElementList {}
 
         public partial class DifferentialComponent : IElementList { }

